bench/algorithm/binarytrees/1.php

```diff
@@ -4,61 +4,41 @@
 
    contributed by Peter Baltruschat
    modified by Levi Cameron
+   modified by Stéphane Demonchaux
    *reset*
+   
+   must run with
+   
+   opcache.jit=1255
+   opcache.enable_cli=1
+   opcache.jit_buffer_size=100M
+   memory_limit=400M
 */
 
-class Node
-{
-    public $l;
-    public $r;
+function checksum(array|null $node):int {
+    return $node === null ? 1 : 1 + checksum($node[0]) + checksum($node[1]);
 }
 
-function bottomUpTree($depth)
-{
-    $node = new Node();
-    if (!$depth) return $node;
-    $depth--;
-    $node->l = bottomUpTree($depth);
-    $node->r = bottomUpTree($depth);
-    return $node;
+function createTree(int $depth): array|null {
+    return $depth-- > 0 ? [createTree($depth), createTree($depth)] : null;
```

Comment:
This is not equivalent to below js code, it halves num of allocations
```javascript
function createTree(depth) {
    return depth-- > 0
        ? { left: createTree(depth), right: createTree(depth) }
        : { left: null, right: null };
}
```

Reply:
done